sql/7_0_2-to-7_0_3_upgrade.sql

```diff
@@ -125,3 +125,116 @@ ALTER TABLE `form_encounter` ADD `ordering_provider_id` INT(11) DEFAULT '0' COMM
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_language', 'Use of language as expressed in the standards in § 170.213', 7);
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_sexual_orientation', 'Use of sexual orientation as expressed in the standards in § 170.213', 8);
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_gender_identity', 'Use of gender identity as expressed in the standards in § 170.213', 9);
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_sex', 'Use of sex as expressed in the standards in § 170.213', 10);
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_dob', 'Use of date of birth as expressed in the standards in § 170.213', 11);
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_sdoh', 'Use of social determinants of health data as expressed in the standards in § 170.213', 12);
+INSERT INTO `list_options` (`list_id`, `option_id`, `title`, `seq`) VALUES ('dsi_evidence_source_attributes', 'evidence_based_health_status', 'Use of health status assessments data as expressed in the standards in § 170.213', 13);
+#EndIf
+
+#IfNotTable dsi_source_attributes
+CREATE TABLE `dsi_source_attributes` (
+     `id` BIGINT(20) UNSIGNED NOT NULL AUTO_INCREMENT,
+     `client_id` VARCHAR(80) NOT NULL,
+     `list_id` VARCHAR(100) NOT NULL,
+     `option_id` VARCHAR(100) NOT NULL,
+     `clinical_rule_id` VARCHAR(31) DEFAULT NULL,
+     `source_value` VARCHAR(2048) NOT NULL,
+     `created_by` BIGINT(20) DEFAULT NULL,
+     `last_updated_by` BIGINT(20) DEFAULT NULL,
+     `created_at` TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+     `last_updated_at` TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+     PRIMARY KEY (`id`),
+     UNIQUE (`list_id`, `option_id`, `client_id`),
+     FOREIGN KEY (`client_id`) REFERENCES `oauth_clients` (`client_id`)
+         ON DELETE RESTRICT
+         ON UPDATE CASCADE,
+     FOREIGN KEY (`list_id`, `option_id`) REFERENCES `list_options` (`list_id`, `option_id`)
+         ON DELETE CASCADE
+         ON UPDATE CASCADE,
+     FOREIGN KEY (`clinical_rule_id`) REFERENCES `clinical_rules` (`id`)
+         ON DELETE SET NULL
+         ON UPDATE CASCADE,
+     FOREIGN KEY (`created_by`) REFERENCES `users` (`id`)
+         ON DELETE RESTRICT
+         ON UPDATE CASCADE,
+     FOREIGN KEY (`last_updated_by`) REFERENCES `users` (`id`)
+         ON DELETE RESTRICT
+         ON UPDATE CASCADE
```

Comment:
can't use foreign keys. we tried in past and it caused problems. couple backup/restore infrastructure issues. biggest as i recall was fragility of encoding/collation (need to have total control of these, which openemr does not have).

Reply:
If needed, could reassess this though. And would need to revisit what happened in past when we did this.

Reply:
Anything that is numeric won't have encoding/collation issues.  I'd prefer to have foreign keys on list_id/option_id but I can drop that if we see it as an issue with collation problems.  

As for backup/restore, issues come about if people don't run the dump tools properly to make sure foreign keys are turned off until the backup is completely restored.

We really need to start making data integrity a safety constraint by default going forward, foreign keys are a good way to do this, but if its going to prevent the PR coming in, I can drop this.

Reply:
Here's a good thread on the issue with encoding/collation which does not seem to be solvable (no way for openemr to enforce consistent collation):
https://github.com/openemr/openemr/issues/4540

Reply:
Dropped the foreign keys, punting for another time.